runtime/onert/backend/cpu/ops/PadLayer.cc

```diff
@@ -27,39 +27,44 @@ namespace cpu
 namespace ops
 {
 
-PadLayer::PadLayer()
-    : _input(nullptr), _output(nullptr), _padData(), _padRank(), _constantValueData()
+PadLayer::PadLayer() : _input(nullptr), _output(nullptr), _padData(), _padRank(), _value()
 {
   // DO NOTHING
 }
 
-void PadLayer::padFloat32()
+template <typename T> void PadLayer::PadImpl(const T *constant_value_data)
 {
-  nnfw::cker::Pad(_padData, _padRank, getTensorShape(_input),
-                  reinterpret_cast<const float *>(_input->buffer()), getTensorShape(_output),
-                  reinterpret_cast<float *>(_output->buffer()), _constantValueData.f);
+  nnfw::cker::Pad<T>(_padData, _padRank, getTensorShape(_input),
+                     reinterpret_cast<const T *>(_input->buffer()), getTensorShape(_output),
+                     reinterpret_cast<T *>(_output->buffer()), constant_value_data);
 }
-void PadLayer::padQuant8() { throw std::runtime_error("Quantized Pad isn't supported NYI"); }
 
 void PadLayer::configure(const IPortableTensor *input, IPortableTensor *output,
-                         const int32_t *padData, int32_t padRank, uint8_t *constantValueData)
+                         const int32_t *padData, int32_t padRank, const void *constantValueData)
 {
   _input = input;
   _output = output;
   memcpy(_padData, padData, sizeof(_padData));
   _padRank = padRank;
-  _constantValueData.u8 = constantValueData;
+  if (_input->data_type() == OperandType::FLOAT32)
+  {
+    _value.f = constantValueData ? *reinterpret_cast<const float *>(constantValueData) : 0.0f;
+  }
+  else if (_input->data_type() == OperandType::QUANT_UINT8_ASYMM)
+  {
+    _value.u8 = constantValueData ? *reinterpret_cast<const uint8_t *>(constantValueData) : 0;
+  }
```

Comment:
Is there any reason to define new `ConstantValue` union and change implementation?

Reply:
In `KernelGenerator.cc` we can read input value to use for PadV2 by `ctx.at(index).data()->base()`.
but it always return const pointer.

In contrast, the 'Dataptr' union used previously is not a const type pointer, so I had a problem between type conversions.
So I define new `ConstantValue` to solve this problem.

Reply:
> If you want to save const pointer, you can define ConstDataPtr uinon in OperationUtils.h and change _constantValueData type to ConstDataPtr. Then we need small change for this implementation.

good suggestion.
let me fix it.